expo-extension-migration.md

```diff
@@ -0,0 +1,101 @@
+# Migration off the expo extension
+
+> TL;DR: Starting in Expo SDK 40, the `.expo.js` extension is deprecated in favor of optional imports.
+
+Files with the `.expo.*` extension will no longer be used in SDK 41+, and the `EXPO_TARGET` environment variable is deprecated.
+
+## Background
+
+In the past, it was impossible to import files that didn't exist using the Metro bundler. Even if you wrapped a require statement in a try/catch, Metro would just fail to bundle instead of throwing. We've upstreamed a fix in SDK 40+ which correctly throws an error:
+
+```ts
+try {
+  // If expo-camera doesn't exist, Metro would just crash instead of throwing :(
+  require("expo-camera");
+} catch (error) {
+  // Starting in SDK 40+ the error will be thrown meaning you can optionally check if modules exist!
+}
+```
+
+To combat this, we created the `.expo.js` extension which would get used in the Expo client or legacy turtle V1 builds (`expo build`). This feature was primarily used by library authors who wanted to support standalone versions of expo packages like `react-native-camera` if `expo-camera` wasn't installed in the project. In SDK 40+ this can be done using try/catch syntax.
```

Comment:
```suggestion
To work around this, we created the `.expo.js` extension which would get used in the Expo client and classic standalone builds (`expo build`). This feature was primarily used by library authors who wanted to support alternative APIs like `react-native-camera` if `expo-camera` wasn't installed in the project. In SDK 40+ this can be done using try/catch syntax.
```